src/EFCore.Relational/Metadata/Conventions/EntityTypeHierarchyMappingConvention.cs

```diff
@@ -0,0 +1,82 @@
+
+namespace Microsoft.EntityFrameworkCore.Metadata.Conventions
+{
+    /// <summary>
+    ///     A convention that removes discriminators from non-TPH entity types and unmaps the inherited properties for TPT entity types.
+    /// </summary>
+    public class EntityTypeHierarchyMappingConvention : IModelFinalizingConvention
+    {
+        /// <summary>
+        ///     Creates a new instance of <see cref="EntityTypeHierarchyMappingConvention" />.
+        /// </summary>
+        /// <param name="dependencies"> Parameter object containing dependencies for this convention. </param>
+        /// <param name="relationalDependencies">  Parameter object containing relational dependencies for this convention. </param>
+        public EntityTypeHierarchyMappingConvention(
+            [NotNull] ProviderConventionSetBuilderDependencies dependencies,
+            [NotNull] RelationalConventionSetBuilderDependencies relationalDependencies)
+        {
+        }
+
+        /// <inheritdoc />
+        public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder, IConventionContext<IConventionModelBuilder> context)
+        {
+            var nonTphRoots = new HashSet<IConventionEntityType>();
+
+            foreach (var entityType in modelBuilder.Metadata.GetEntityTypes())
+            {
+                if (entityType.BaseType == null)
+                {
+                    continue;
+                }
+
+                var tableName = entityType.GetTableName();
+                var schema = entityType.GetSchema();
+                if (tableName != null
+                    && (tableName != entityType.BaseType.GetTableName()
+                        || schema != entityType.BaseType.GetSchema()))
+                {
+                    nonTphRoots.Add(entityType.GetRootType());
+                    foreach (var property in entityType.BaseType.GetProperties())
+                    {
+                        if (property.IsPrimaryKey())
+                        {
+                            continue;
+                        }
+
+                        property.SetColumnName(null, tableName, schema);
+                    }
+                }
+
+                var viewName = entityType.GetViewName();
+                var viewSchema = entityType.GetViewSchema();
+                if (viewName != null
+                    && (viewName != entityType.BaseType.GetViewName()
+                        || viewSchema != entityType.BaseType.GetViewSchema()))
+                {
+                    nonTphRoots.Add(entityType.GetRootType());
+                    foreach (var property in entityType.BaseType.GetProperties())
+                    {
+                        if (property.IsPrimaryKey())
+                        {
+                            continue;
+                        }
+
+                        property.Builder.HasViewColumnName(null, viewName, viewSchema);
```

Comment:
```suggestion
                        property.SetViewColumnName(null, viewName, viewSchema);
```

to match `property.SetColumnName()` above.